server/webapp/WEB-INF/rails/webpack/models/config_repos/config_repos_crud.ts

```diff
@@ -17,14 +17,18 @@
 import {ApiRequestBuilder, ApiResult, ApiVersion, ObjectWithEtag} from "helpers/api_request_builder";
 import SparkRoutes from "helpers/spark_routes";
 import {ConfigRepoJSON, ConfigReposJSON} from "models/config_repos/serialization";
-import {ConfigRepo, ConfigRepos} from "models/config_repos/types";
+import {ConfigRepo, ConfigRepos, Material} from "models/config_repos/types";
 
 const s = require("helpers/string-plus");
 
-export function toSnakeCaseJSON(o: ConfigRepo) {
-  const configurations = o.createConfigurationsFromText();
+function toSnakeCaseJSON(o: object) {
   const text           = JSON.stringify(o, s.snakeCaser);
-  const json           = JSON.parse(text);
+  return JSON.parse(text);
+}
+
+export function configRepoToSnakeCaseJSON(o: ConfigRepo) {
+  const configurations = o.createConfigurationsFromText();
+  const json           = toSnakeCaseJSON(o);
```

Comment:
I can see why this has been done. Maybe add a comment why?

Reply:
The intention was to separate the logic of converting object to JSON (`toSnakeCaseJSON ()`) and config repo specific logic. So that `toSnakeCaseJSON()` can be reused for other objects like `Material`